test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,459 @@
+	if err != nil {
+		t.Fatalf("failed to create nginx ConfigMap: %v", err)
+	}
+	require.NoError(t, err)
+
+	deploymentLabels := map[string]string{
+		"app": "nginx",
+	}
+
+	nginxDeployObject, err := data.CreateNginxDeploymentForTest("nginx-deployment", data.testNamespace, nginxConfiMapObject.Name, 2, deploymentLabels)
+	if err != nil {
+		t.Fatalf("failed to create nginx deployment: %v", err)
+	}
+	require.NoError(t, err)
+
+	// Though this is used in vmagent_test.go but i think i needed it here to check for the deployment.
+	//TODO: Time of 15 seconds is still large for a timeout.
+	err = data.waitForDeploymentReady(t, data.testNamespace, nginxDeployObject.Name, 15*time.Second)
+	if err != nil {
+		t.Fatalf("error while waiting for nginx deployment to be ready : %v", err)
+	}
+	require.NoError(t, err)
+
+	k8sUtils, err = NewKubernetesUtils(data)
+	if err != nil {
+		t.Fatalf("error getting k8s utils %+v", err)
+	}
+	require.NoError(t, err)
+
+	nginxPods, err := k8sUtils.GetPodsByLabel(data.testNamespace, "app", "nginx")
+	if err != nil {
+		t.Fatalf("error getting Pods by label  %+v", err)
+	}
+	require.NoError(t, err)
+
+	//domainMapping holds whether the IP is mapped to Domain or not.
+	domainMapping := make(map[string]bool)
+
+	// pick an IP to be added in config
+	var ipForConfig string
+	for idx, pod := range nginxPods {
+		//TODO: Following wait time change ?
+		_, err = data.podWaitForIPs(10*time.Second, pod.Name, data.testNamespace)
+		if err != nil {
+			t.Fatalf("error waiting for nginx pods to get IPs %+v", err)
+		}
+		require.NoError(t, err)
+
+		ipStr := strings.TrimSpace(pod.Status.PodIP)
+		domainMapping[ipStr] = false
+		//pick last IP for config
+		if idx == len(nginxPods)-1 {
+			ipForConfig = ipStr
+		}
+	}
+
+	// Create and update the custom DNS configMap
+	customDNSconfig := fmt.Sprintf(`lfx.test:53 {
+    errors
+    t
+    health
+    kubernetes cluster.local in-addr.arpa ip6.arpa {
+        pods insecure
+        fallthrough in-addr.arpa ip6.arpa
+        ttl 60
+    }
+    hosts {
+        %s %s
+        no_reverse
+        pods verified
+        ttl 10
+    }
+    loop
+    reload
+}`, ipForConfig, testFqdn)
+
+	customDNSconfigData := map[string]string{
+		"Corefile": customDNSconfig,
+	}
+
+	customDNSconfigMapObject, err := data.CreateConfigMap(data.testNamespace, "custom-dns-config", customDNSconfigData, nil, false)
+
+	if err != nil {
+		t.Fatalf("failed to create custom dns ConfigMap: %v", err)
+	}
+	require.NoError(t, err)
+	domainMapping[ipForConfig] = true
+
+	// create supporting SA, Role and Role Binding for DNS deployment.
+	saSpec := data.BuildServiceAccount("custom-dns-service-account", data.testNamespace, nil)
+	sa, err := data.CreateOrUpdateServiceAccount(saSpec)
+	if err != nil {
+		t.Fatalf("failed to create service acount for custom dns : %v", err)
+	}
+	require.NoError(t, err)
+
+	clusterRoleSpec := &v12.ClusterRole{
```

Comment:
all of this should not be necessary for out use case: we don't need to monitor K8s resources, and you should be able to just remove the `kubernetes` entry from your custom CoreDNS configuration?